From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
Date: Thu, 29 May 2025 19:31:51 +0300
Subject: [WIP-FRL] arm64: dts: rockchip: Add tmds-enable-gpios to
 rk3588-rock-5b

In preparation to support HDMI 2.1 FRL operating mode, make use of the
GPIO4_B1 and GPIO4_A1 lines (labeled HDMI0_TX_ON_H & HDMI1_TX_ON_H in
the schematics) required to control the level shifters for HDMI0 and
HDMI1 data lines, respectively.

The lines will be asserted when operating in TMDS mode and deasserted
when switching to FRL.

Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
---
 arch/arm64/boot/dts/rockchip/rk3588-rock-5b-5bp-5t.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b-5bp-5t.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b-5bp-5t.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b-5bp-5t.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b-5bp-5t.dtsi
@@ -183,6 +183,7 @@ &gpu {
 };
 
 &hdmi0 {
+	tmds-enable-gpios = <&gpio4 RK_PB1 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 };
 
@@ -205,6 +206,7 @@ &hdmi0_sound {
 &hdmi1 {
 	pinctrl-0 = <&hdmim0_tx1_cec &hdmim0_tx1_hpd
 		     &hdmim1_tx1_scl &hdmim1_tx1_sda>;
+	tmds-enable-gpios = <&gpio4 RK_PA1 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 };
 
-- 
Armbian

